### PR TITLE
[MM-49260] Fixing accessibility issues for insights

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
I was playing around with axe-core DevTools and it picked up a few minor accessibility issues for insights

1. Sidebar menu items roles have incorrect parents, move the role='listitem' up to `<li>`
2. Modal buttons have no aria labels
3. least active channels action buttons need aria labels
4. aria-readonly on incorrect role for top threads

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49260

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Release Note

```release-note
NONE
```

[MM-49260]: https://mattermost.atlassian.net/browse/MM-49260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kostaspt — APPROVED** [review]

File: components/activity_and_insights/insights/card/card.tsx [hunk 1]
```diff
@@ -48,6 +51,15 @@ const InsightsCard = (props: Props) => {
...
+                    aria-label={formatMessage(
+                        {
+                            id: 'insights.card.ariaLabel', 
+                            defaultMessage: 'Open modal for {title}'
+                        },
+                        {
+                            title: props.title,
+                        }
+                    )}
                 >
```
Comment (cyrilzhang-mm) [inline components/activity_and_insights/insights/card/card.tsx:63]: Maybe add `aria-haspopup={true}`?

File: components/activity_and_insights/activity_and_insights_link/activity_and_insights_link.tsx [hunk 2]
```diff
@@ -41,14 +41,14 @@ const ActivityAndInsightsLink = () => {
                     active: inInsights,
                 })}
                 tabIndex={-1}
+                role='listitem'
```
Comment (cyrilzhang-mm) [inline components/activity_and_insights/activity_and_insights_link/activity_and_insights_link.tsx:44]: I don't see why we need to have `role='listitem'` on a `li` element? It should be implicit.

**Review by cyrilzhang-mm — APPROVED** [review]
Minor comments, LGTM otherwise

File: components/activity_and_insights/insights/least_active_channels/channel_actions_menu/channel_actions_menu.tsx [hunk 3]
```diff
@@ -70,7 +70,10 @@ const ChannelActionsMenu = ({channel, actionCallback}: Props) => {
                 stopPropagationOnToggle={true}
                 id={`customWrapper-${channel.id}`}
             >
-                <button className='icon action-wrapper'>
+                <button 
+                    className='icon action-wrapper'
+                    aria-label={localizeMessage('insights.leastActiveChannels.menuButtonAriaLabel', 'Open manage channel menu')}
```
Comment (M-ZubairAhmed) [inline components/activity_and_insights/insights/least_active_channels/channel_actions_menu/channel_actions_menu.tsx:75]: can we just use formatMessage here?

File: components/activity_and_insights/insights/card/card.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply components/activity_and_insights/insights/card/card.tsx:63]: isnt that attribute used for popover instead of modal?

Comment (BenCookie95) [reply components/activity_and_insights/insights/card/card.tsx:63]: It's used for a modal as well

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]

**Pull request merged**